lox_services/persistence/database/query_handlers.py

```diff
@@ -2,26 +2,36 @@
 import os
 import re
 import time
+from typing import Any, Optional, Tuple
+from collections.abc import Sequence
 
-from google.cloud.bigquery import Client
+from google.cloud.bigquery import Client, QueryJobConfig, ScalarQueryParameter, ArrayQueryParameter
 from google.cloud.bigquery.job import QueryJob
 from pandas import DataFrame
 
 from lox_services.persistence.config import SERVICE_ACCOUNT_PATH
 from lox_services.persistence.database.exceptions import BadQueryTypeException, MissingUpdateDatetimeException
 import lox_services.utils.general_python as gpy
-from lox_services.utils.enums import Colors
+from lox_services.utils.enums import BQParameterType, Colors
 from lox_services.utils.metadata import get_function_callers
 
-def raw_query(query: str, print_query: bool = True) -> QueryJob:
+def raw_query(
+    query: str, 
+    *, 
+    parameters:  Optional[Sequence[Tuple[str, BQParameterType, Any]]] = None,
+    print_query: bool = True
+    ) -> QueryJob:
+
     """Excecutes a query with Google BigQuery, without any checks.
         Adds some metadata at the beginning of the query.
         ## Arguments
         - `query`: String representation of the query to be executed.
         - `print_query`: Tells whether or not to print the query before executing it.
+        - `parameters`: list of parameter #used to avoid sql injection
 
         ## Example
             >>> raw_query("SELECT * FROM InvoicesData.Refunds LIMIT 10")
+            >>> select ("SELECT * FROM InvoicesData.Refunds where carrier=@carrier LIMIT 10", parameters = [("carrier", "STRING", "UPS")])
```

Comment:
If your type hints are correct, then this code would not work. 

 `parameters = [("carrier", BQParameterType.STRING, "UPS")])`